File: effective-java/src/main/java/com/example/effectivejava/chapter01/item01/AppConfig.java
package com.example.effectivejava.chapter01.item01;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class AppConfig {

    @Bean
    public HelloService helloService() {
        return new ChineseHelloService();
    }
}

File: spring-ai/openai/src/main/kotlin/com/example/openai/config/AppConfig.kt
package com.example.openai.config

import org.springframework.ai.chat.client.ChatClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class AppConfig {

    @Bean
    fun chatClient(chatClientBuilder: ChatClient.Builder): ChatClient {
        return chatClientBuilder
            .defaultSystem("당신은 교육 튜터입니다. 개념을 명확하고 간단하게 설명하세요")
            .build()
    }
}